### PR TITLE
Create, start and stop application workers with their transport workers

### PR DESCRIPTION
**Pull request by justinvdm**
#20 adds an application worker for `POST`ing mo messages to a url. We plan to have one application worker instance per transport instance, and create, start and stop each application worker with its corresponding transport worker.

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review, though I still need to figure out why the application worker service isn't being run.

File: junebug/tests/test_channel.py [hunk 1]
```diff
@@ -72,6 +91,43 @@ def test_update_channel(self):
...
+        self.assertEqual(self.service.namedServices[channel.id], worker1)
+        yield channel.update({'foo': 'bar'})
+        self.assertEqual(self.service.namedServices[channel.id], worker1)
+
+        config = self.create_channel_config()
+        config['config']['foo'] = ['bar']
+        yield channel.update(config)
+
+        worker2 = channel.transport_worker
+        self.assertEqual(self.service.namedServices[channel.id], worker2)
```
Comment (rudigiesler) [inline junebug/tests/test_channel.py]: Should we maybe assert that `worker1` is no longer in the list of services?

File: junebug/tests/test_channel.py [hunk 2]
```diff
@@ -72,6 +91,43 @@ def test_update_channel(self):
...
+        yield channel.update({'foo': 'bar'})
+        self.assertEqual(self.service.namedServices[id], worker1)
+
+        config = self.create_channel_config()
+        config['config']['foo'] = ['bar']
+        yield channel.update(config)
+
+        worker2 = channel.application_worker
+        self.assertEqual(self.service.namedServices[id], worker2)
+        self.assertNotEqual(worker1, worker2)
```
Comment (rudigiesler) [inline junebug/tests/test_channel.py]: Shouldn't we also be restarting the application if the `mo_url` attribute changes? Not only if the `config` (which is the transport config) changes.

**Comment by rudigiesler** [comment]
Just those two comments from me, looks good otherwise

File: junebug/tests/test_channel.py (hunk 2 above)
Comment (justinvdm) [inline junebug/tests/test_channel.py]: Good point, will change.

**Comment by justinvdm** [comment]
Ready for re-review.

**Comment by rudigiesler** [comment]
:+1:

**Pull request merged**